### PR TITLE
Change status from progress

### PR DESCRIPTION
**Pull request by arek1-618**
Change status from progress

### CODE REVIEW

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 1]
```diff
@@ -5,10 +5,12 @@
...
 				{assign var=PICKLIST_VALUES value=\App\Fields\Picklist::getValues($NAME)}
 				<div class="tpl-Base-Detail-HeaderProgress c-arrows px-3 w-100">
-					<ul class="c-arrows__container">
+					<ul class="c-arrows__container js-header-progress-bar" data-picklist-name="{$NAME}"
+						data-js="container">
 						{assign var=ARROW_CLASS value="before"}
 						{foreach from=$PICKLIST_VALUES item=VALUE_DATA name=picklistValues}
-							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}">
+							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}"
+								data-picklist-value="{$VALUE_DATA['picklistValue']}" data-js="confirm|click">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: There are no conditions for editing the values

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 2]
```diff
@@ -2454,7 +2485,8 @@ jQuery.Class("Vtiger_Detail_Js", {
 			tabElement.data('url', url);
 			tabElement.trigger('click');
 		});
-	},
+	}
+	,
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: ?

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 3]
```diff
@@ -3,12 +3,19 @@
 	{if isset($FIELDS_HEADER['progress'])}
 		{foreach from=$FIELDS_HEADER['progress'] key=NAME item=FIELD_MODEL}
 			{if !$RECORD->isEmpty($NAME)}
+				{\App\DebugerEx::log(Users_Record_Model::getCurrentUserModel()->get('roleid'))}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: ??

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 4]
```diff
@@ -3,12 +3,19 @@
...
 				{assign var=PICKLIST_VALUES value=\App\Fields\Picklist::getValues($NAME)}
+				{assign var=PICKLIST_VALUES_BY_ROLE value=\App\Fields\Picklist::getRoleBasedPicklistValues($NAME, Users_Record_Model::getCurrentUserModel()->get('roleid'))}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: Users_Record_Model => $USER_MODEL

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 5]
```diff
@@ -3,12 +3,19 @@
...
+					<ul class="c-arrows__container js-header-progress-bar" data-picklist-name="{$NAME}"
+						data-js="container">
 						{assign var=ARROW_CLASS value="before"}
 						{foreach from=$PICKLIST_VALUES item=VALUE_DATA name=picklistValues}
-							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}">
+							<li
+									class="c-arrows__item{if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}"
+									data-picklist-value="{$VALUE_DATA['picklistValue']}"
+									data-js="confirm|click"
+									data-access="{if in_array($VALUE_DATA['picklistValue'], $PICKLIST_VALUES_BY_ROLE)}1{else}0{/if}">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: $FIELD_MODEL->isEditable()

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 6]
```diff
@@ -4,11 +4,16 @@
 		{foreach from=$FIELDS_HEADER['progress'] key=NAME item=FIELD_MODEL}
 			{if !$RECORD->isEmpty($NAME)}
 				{assign var=PICKLIST_VALUES value=\App\Fields\Picklist::getValues($NAME)}
+				{assign var=PICKLIST_VALUES_BY_ROLE value=\App\Fields\Picklist::getRoleBasedPicklistValues($NAME, $USER_MODEL->get('roleid'))}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: not always

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 7]
```diff
@@ -2191,6 +2191,48 @@ jQuery.Class("Vtiger_Detail_Js", {
...
+	},
+	/**
+	 * Change status from progress
+	 */
+	registerProgress() {
+		const thisInstance = this;
+		let progressList = $('.js-header-progress-bar').eq(0);
+		if (progressList.length > 0) {
+			let picklistName = progressList.data('picklistName');
+			progressList.find('li').each((index, element) => {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: Skip unmodified values

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 8]
```diff
@@ -4,11 +4,14 @@
 		{foreach from=$FIELDS_HEADER['progress'] key=NAME item=FIELD_MODEL}
 			{if !$RECORD->isEmpty($NAME)}
 				{assign var=PICKLIST_VALUES value=\App\Fields\Picklist::getValues($NAME)}
+				{if $FIELD_MODEL->get('uitype')===15}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: PICKLIST_VALUES = $FIELD_MODEL->getPicklistValues()

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 9]
```diff
@@ -4,11 +4,14 @@
...
+					{assign var=PICKLIST_VALUES_BY_ROLE value=\App\Fields\Picklist::getRoleBasedPicklistValues($NAME, $USER_MODEL->get('roleid'))}
+				{/if}
 				<div class="tpl-Base-Detail-HeaderProgress c-arrows px-3 w-100">
-					<ul class="c-arrows__container">
+					<ul class="c-arrows__container js-header-progress-bar" data-picklist-name="{$NAME}" data-js="container">
 						{assign var=ARROW_CLASS value="before"}
 						{foreach from=$PICKLIST_VALUES item=VALUE_DATA name=picklistValues}
-							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}">
+							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}{if $RECORD->isEditable() && $FIELD_MODEL->isAjaxEditable() && ($FIELD_MODEL->get('uitype')!==15 || in_array($VALUE_DATA['picklistValue'], $PICKLIST_VALUES_BY_ROLE))} js-access{/if}" data-picklist-value="{$VALUE_DATA['picklistValue']}" data-js="confirm|click">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: remove validations by uitype

File: languages/pl_pl/_Base.json [hunk 10]
```diff
@@ -1433,6 +1433,7 @@
     "JS_YOU_ARE_IN_PAGE_NUMBER": "Jeste\u015b na stronie numer",
     "JS_YOU_CAN_SELECT_ONLY": "Mo\u017cna wybra\u0107 tylko",
     "LBL_DELETE_CONFIRMATION": "Czy na pewno chcesz usun\u0105\u0107 rekord?",
+    "JS_CHANGE_STATUS_CONFIRMATION": "Czy chcesz ustawi\u0107 ten status?",
```
Comment (rskrzypczak) [inline languages/pl_pl/_Base.json]: Change label

File: layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl [hunk 11]
```diff
@@ -1,14 +1,19 @@
...
 				{assign var=PICKLIST_VALUES value=\App\Fields\Picklist::getValues($NAME)}
-				<div class="tpl-Base-Detail-HeaderProgress c-arrows px-3 w-100">
-					<ul class="c-arrows__container">
+				<div class="c-arrows px-3 w-100">
+					<ul class="c-arrows__container js-header-progress-bar" data-picklist-name="{$NAME}"
+						data-js="container">
 						{assign var=ARROW_CLASS value="before"}
 						{foreach from=$PICKLIST_VALUES item=VALUE_DATA name=picklistValues}
-							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}">
+							<li class="c-arrows__item {if $smarty.foreach.picklistValues.first}first{/if} {if $VALUE_DATA['picklistValue'] eq $RECORD->get($NAME)}active{assign var=ARROW_CLASS value="after"}{else}{$ARROW_CLASS}{/if}{if $RECORD->isEditable() && $FIELD_MODEL->isAjaxEditable() && $VALUE_DATA['picklistValue'] !== $RECORD->get($NAME) && isset($PICKLIST_OF_FIELD[$VALUE_DATA['picklistValue']])} js-access{/if}"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/HeaderProgress.tpl]: isset($PICKLIST_OF_FIELD[$VALUE_DATA['picklistValue']]) not required

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 12]
```diff
@@ -2191,6 +2191,43 @@ jQuery.Class("Vtiger_Detail_Js", {
 			summaryView.hide();
 		});
 	},
+	/**
+	 * Show confirmation on event click
+	 * @param jQuery element
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: {jQuery}

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 13]
```diff
@@ -2191,6 +2191,43 @@ jQuery.Class("Vtiger_Detail_Js", {
...
+	 * @param string picklistName
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: {string}

**Pull request merged**